### PR TITLE
Simplify conditional branch in _read function in readers.py

### PR DESCRIPTION
**Pull request by Compro-Prasad**
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -542,13 +542,9 @@ def _read(
...
     # if we pass a date_parser and parse_dates=False, we should not parse the
     # dates GH#44366
-    if (
-        kwds.get("date_parser", None) is not None
-        and kwds.get("parse_dates", None) is None
-    ):
-        kwds["parse_dates"] = True
-    elif kwds.get("parse_dates", None) is None:
-        kwds["parse_dates"] = False
+    if kwds.get("parse_dates", None) is None:
```
Comment (phofl) [inline pandas/io/parsers/readers.py:545]: I prefer the old version, more readable imo

Comment (Compro-Prasad) [reply pandas/io/parsers/readers.py:545]: Readability IMO is subjective. What might be readable to me might be unreadable to someone else.

Anyway, I would consider the following more readable:

```
if kwds.get("parse_dates", None) is None:
    if kwds.get("date_parser", None) is not None:
        kwds["parse_dates"] = True
    else:
        kwds["parse_dates"] = False
```

Would like to know your opinion on the above.

Comment (phofl) [reply pandas/io/parsers/readers.py:545]: Yep this is better

**Review by mroeschke — APPROVED** [review]

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @Compro-Prasad

**Pull request merged**